packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -38,69 +44,92 @@ export const AuthModal = ({
     onLoginSuccess = noop,
     onRegistrationSuccess = noop,
     onPasswordResetSuccess = noop,
+    isOpen,
+    onOpen,
+    onClose,
     ...props
 }) => {
     const {formatMessage} = useIntl()
-    const customer = useCustomer()
+    const customerId = useCustomerId()
+    const customerType = useCustomerType()
+    const customer = useCustomer(
+        {customerId},
+        {enabled: !!customerId && customerType === 'registered'}
+    )
```

Comment:
This endpoint has a quirk, it returns 401 if you are a guest user. It's very confusing. Should we handle this quirk at the SDK level to always have `{enabled: customerType === 'registered'}`? 🤔

Reply:
I mean if "get customer" is actually "get (registered) customer", then the 404 makes sense. I'd say yeah, should probably do that.

Reply:
We talked about the challenge last week, hence i created a ticket to investigate if we can export the required fields metadata from isomorphic library. then we can have a generic solution: `{enabled: client.method.requiredFields}`